network/response_parser.c

```diff
@@ -20,80 +22,100 @@ int parse_uri(char *response, http_response *result);
  * @param index
  * @return index of body start in response
  */
-int parse_header(char *response, http_response *result, int index);
+int parse_header(char *line, http_response *result, int index);
 
 /**
  * parse http response to
  * @param response
  * @return http response
  */
 http_response *parse(char *response) {
-	http_response *result = malloc(sizeof(http_response) + 4 * sizeof(char *));
-	int header_index = parse_uri(response, result);
-	int body_index = parse_header(response, result, header_index);
-	result->body = string_from_to(response, body_index, length_pointer_char(response) - 1);
-	return result;
+    char** response_split = string_split(response, "\r\n");
+    int response_split_length = length_pointer_pointer_char(response_split);
+    http_response *result = malloc(sizeof(http_response) + 4 * sizeof(char *));
+    result->header_quantity = count_header(response);
+    if (response_split_length < 2) {
+        free_pointer_pointer_char(response_split);
+        return result;
+    }
+
+    if (result->header_quantity == 0 || !parse_uri(response_split[0], result)) {
+        return result;
```

Comment:
This function has been changed, please review the newest [Line 51](https://github.com/foodtiny/native/pull/77/commits/e410a58f31aa0a6944db5dd27541266a548ccaf2?diff=unified#diff-24bfe6d8deab1780cdbdee64b32562eaL32)